RUMM-1956 Document all regions in setup instruction (#749)
DataDog/dd-sdk-ios: docs/trace_collection.md

[builder setWithEndpoint:[DDEndpoint us5]];

[DDDatadog initializeWithAppContext:[DDAppContext new]
                    trackingConsent:trackingConsent
                      configuration:[builder build]];
```
{{% /tab %}}
{{< /tabs >}}
{{< /site-region >}}

{{< site-region region="gov" >}}
{{< tabs >}}
{{% tab "Swift" %}}
```swift
Datadog.initialize(
    appContext: .init(),
    trackingConsent: trackingConsent,
    configuration: Datadog.Configuration
        .builderUsing(clientToken: "<client_token>", environment: "<environment_name>")
        .set(serviceName: "app-name")
        .set(endpoint: .us1_fed)
        .build()
)
```
{{% /tab %}}
{{% tab "Objective-C" %}}
```objective-c
DDConfigurationBuilder *builder = [DDConfiguration builderWithClientToken:@"<client_token>"
                                                              environment:@"<environment_name>"];
[builder setWithServiceName:@"app-name"];
[builder setWithEndpoint:[DDEndpoint us1_fed]];

[DDDatadog initializeWithAppContext:[DDAppContext new]
                    trackingConsent:trackingConsent
                      configuration:[builder build]];
```
{{% /tab %}}
{{< /tabs >}}
{{< /site-region >}}

> To be compliant with the GDPR regulation, the SDK requires the `trackingConsent` value at initialization.
> The `trackingConsent` can be one of the following values:

Reviewer: ```suggestion
To comply with GDPR regulations, the SDK requires the `trackingConsent` value at initialization.

Use one of the following values for `trackingConsent`:
```

> - `.pending` - the SDK starts collecting and batching the data but does not send it to Datadog. The SDK waits for the new tracking consent value to decide what to do with the batched data.
> - `.granted` - the SDK starts collecting the data and sends it to Datadog.
> - `.notGranted` - the SDK does not collect any data: logs, traces, and RUM events are not sent to Datadog.

Reviewer: ```suggestion
- `.pending` - the SDK starts collecting and batching the data but does not send it to Datadog. The SDK waits for the new tracking consent value to decide what to do with the batched data.
- `.granted` - the SDK starts collecting the data and sends it to Datadog.
- `.notGranted` - the SDK does not collect any data. No logs, traces, or RUM events are sent to Datadog.
```

To change the tracking consent value after the SDK is initialized, use the `Datadog.set(trackingConsent:)` API call.
The SDK changes its behavior according to the new value. For example, if the current tracking consent is `.pending`:

> - if changed to `.granted`, the SDK will send all current and future data to Datadog;
> - if changed to `.notGranted`, the SDK will wipe all current data and will not collect any future data.

Reviewer: ```suggestion
- If changed to `.granted`, the SDK sends all current and future data to Datadog.
- If changed to `.notGranted`, the SDK wipes all current data and does not collect future data.
```
